app/javascript/components/DownloadButton.js

```diff
@@ -0,0 +1,133 @@
+
+  return [authCode, timeInterval, downloadCommand];
+}
+
+function DownloadCommandContainer() {
+  
+  const [authCode, timeInterval, downloadCommand] = fetchDownloadConfig();
+
+  const expiresMinutes = Math.floor(timeInterval / 60); // seconds -> minutes
+
+  const commandID = 'command-' + authCode;
+
+  return (
+    <div>
+      <div class="input-group">
+        <input
+          id={commandID}
+          class="form-control curl-download-command"
+          value={downloadCommand}
+          readOnly
+        />
+        <span class="input-group-btn"> +
+          <button 
+            id={'copy-button-' + authCode}
+            class="btn btn-default btn-copy" 
+            data-clipboard-target={'#' + commandID}
+            data-toggle="tooltip"
+            title="Copy to clipboard">
+            <i class="far fa-copy"></i>
+          </button>
+          <button 
+            id={'refresh-button-' + authCode}
+            class="btn btn-default btn-refresh glyphicon glyphicon-refresh"
+            data-toggle="tooltip"
+            title="Refresh download command">
+          </button>
+        </span>
+        </div>
+      <div style={{fontSize: '12px'}}>
+        Valid for one use within {' '}
+        <span class="countdown" id={'countdown-' + authCode}>
+          {expiresMinutes}
+        </span>{' '}
+        minutes.  If your command has expired, click refresh button at right in this box.
+      </div>
+    </div>
+  );
+}
+
+
+/**
+ * Component for "Download" button and Bulk Download modal.
+ *
+ * UI spec: https://projects.invisionapp.com/d/main#/console/19272801/402387755/preview
+ */
+export default function DownloadButton(props) {
+
+  const [show, setShow] = useState(false);
+  const [showDownloadCommand, setShowDownloadCommand] = useState(false);
+
+  function showModalAndFetchDownloadCommand() {
+    setShow(!show);
+  }
+
+  const handleClose = () => setShow(false);
+  const handleShow = () => setShow(true);
+
+  window.SCP.initBulkDownloadClipboard();
+
+  return (
+      <>
+      <span
+        id='download-button'
+        className={`${show ? 'active' : ''}`}>
+        <span
+          onClick={showModalAndFetchDownloadCommand}>
+          <FontAwesomeIcon className="icon-left" icon={faDownload}/>
+          Download
+        </span>
+      </span>
+      <Modal
+        id='bulk-download-modal'
+        show={show}
+        onHide={handleClose}
+        animation='false'
+        bsSize='large'
+      >
+        <Modal.Header closeButton>
+          <Modal.Title><h2 class='text-center'>Bulk Download</h2></Modal.Title>
+        </Modal.Header>
+  
+        <Modal.Body>
+          <p className='lead'>
+          To download files matching your search, copy this command and paste it into your terminal:
+          </p>
+          <p className='lead command-container' id='command-container-all'>
+            <DownloadCommandContainer />
+          </p>
+        </Modal.Body>
+
+      </Modal>
+      </>
+    );
+}
```

Comment:
You did a really good job of scoping and structuring this file so it was easy to read -- thanks!